Assets/Scripts/SunComponent.cs

```diff
@@ -1,50 +1,129 @@
     }
 
     // Update is called once per frame
     void Update()
     {
-
+        transform.LookAt(new Vector3(0,0,0));
     }
 
     public void CursorHighlightSun(bool highlight)
     {
         if (highlight)
         {
-            transform.localScale = sceneInitialScale * scaleFactor;
+            SunVisible.transform.localScale = sceneInitialScale * scaleFactor;
         } else
         {
-            transform.localScale = sceneInitialScale;
+            SunVisible.transform.localScale = sceneInitialScale;
         }
     }
 
+    // this is called when another object is selected
+    private void StarSelected(Star selectedStarData, string playerName, Color playerColor)
+    {
+        Destroy(FloatingInfoPanel);
+    }
+    private void SunSelected(bool selected)
+    {
+        Destroy(FloatingInfoPanel);
+    }
+    private void MoonSelected(bool selected)
+    {
+        Destroy(FloatingInfoPanel);
+    }
+
     public void HandleSelectSun(bool broadcastToNetwork = false)
     {
         if (!menuController) menuController = FindObjectOfType<MenuController>();
         if (!menuController.IsDrawing)
         {
             CCDebug.Log("Selected sun", LogLevel.Info, LogMessageCategory.Interaction);
             SimulationEvents.Instance.SunSelected.Invoke(true);
+
+            // make a new floating info panel that is a child of the object
+            FloatingInfoPanel = Instantiate(FloatingInfoPanelPrefab, new Vector3(0, -8f, 6f), new Quaternion(0, 0, 0, 0), this.transform);
+            FloatingInfoPanel.transform.localPosition = new Vector3(0, -9f, 5f);
+            FloatingInfoPanel.transform.localScale = new Vector3(.8f, .8f, .8f);
+
+            setSunSelectedText();
         }
     }
 
+    private void setSunSelectedText()
+    {
+        var solarPosition = SunCalc.GetSunPosition(manager.CurrentSimulationTime, manager.CurrentLatLng.Latitude, manager.CurrentLatLng.Longitude);
+        double azimuth = 180f + (solarPosition.Azimuth * 180f / Mathf.PI);
+        double altitude = solarPosition.Altitude * 180f / Mathf.PI;
+        double declination = solarPosition.Declination * 180f / Mathf.PI;
+        double ra = solarPosition.RA;
+        double raInDegrees = ra * 180f / Mathf.PI;
+        double raInHours = raInDegrees / 15f;
+        if (raInHours < 0) raInHours = 24f + raInHours;
+        StringBuilder description = new StringBuilder();
+        description.AppendLine("The Sun");
+        description.Append("Alt/Az: ")
+            .Append(altitude.ToString("F2"))
+            .Append("°, ")
+            .Append(azimuth.ToString("F2"))
+            .AppendLine("°");
+        description.Append("R.A.: ")
+            .Append(raInHours.ToString("F2"))
+            .Append("h  Dec: ")
+            .Append(declination.ToString("F2"))
+            .AppendLine("°");
+        FloatingInfoPanel.GetComponent<FloatingInfoPanel>().InfoText.SetText(description.ToString());
```

Comment:
I am not sure if C# has anything like mixins / modules / partial interface implementations, but it seems like many of these celestial object components have very similar methods for:

1. listening to events from other celestial objects (selected/deselected)
2. Instantiation of the FloatingPanel prefab
3. calculating alt/az/dec of the object
4. handling deselection

This is obviously NBD, but if it were possible to share functionality, you might end up saving time over the long term.